cmd/falco/runner.go

```diff
@@ -112,6 +119,16 @@ func NewRunner(rz Resolver, c *Config) (*Runner, error) {
 		}()
 	}
 
+	if f != nil {
+		ctx, timeout := _context.WithTimeout(_context.Background(), 20*time.Second)
+		defer timeout()
+		snippets, err := NewSnippet(f).Fetch(ctx)
+		if err != nil {
+			writeln(red, err.Error())
+		}
+		r.snippets = append(r.snippets, snippets...)
+	}
+
```

Comment:
If I understand correctly the behavior here. We are trying to load the runner with snippets, backends, ACLS, and Dicts from either remote or Fastly depending on the config.

I think sharing the same Interface between Terraform "Fetcher" and Fastly Fetcher doesn't make sense in my opinion and you can tell from the `context` object that you are passing to the Terraform fetcher without the need for it! 
I don't believe we need to unify the 2 interfaces. we can simply pass them to the runner before. We can do a Fastly call before reaching the `runner` and let the runner takes it from there! 

I don't think the runner should take care of the behavior or filling these objects

Wdyt?

Reply:
simplified the interface by taking the context in the constructor via a timeout